vignettes/pander.Rmd

```diff
@@ -0,0 +1,197 @@
+
+* [slides on "Building a reporting sytem with BREW"](http://www.slideshare.net/xavierguardiola/building-a-reporting-sytem-with-brew)
+* [learnr blogpost on brew](https://learnr.wordpress.com/2009/09/09/brew-creating-repetitive-reports/)
+
+A brew document is a simple text file with some special tags. `Pandoc.brew` uses only two of them (as building on a personalized version of Jeff's really great brew function):
+
+* `<% ... %>` stand for running inline R commands as usual,
+* `<%= ... %>` does pretty much the same but applies pander to the returning R object (instead of cat like the original brew function does). So putting there any R object, it would return in a nice Pandoc's markdown format with all possible error/warning messages etc.
+
+This latter tries to be smart in some ways:
+
+* A code chunk block (`R` commands between the tags) can return any number of values at any part of the block.
+* Plots and images are grabbed in the document, rendered to a png file and pander method would result in a Pandoc markdown formatted image link. This means that the image would be rendered/shown/included in the exported document.
+* All warnings/messages and errors are recorded in the blocks and returned in the document as footnotes or inline messages.
+* All heavy `R` commands (e.g. those taking more then 0.1 sec to evaluate) are cached so rebrewing a report would not result in a coffee break.
+
+Besides this, the custom brew function can do more and also less compared to the original brew package. First of all, the internal caching mechanism of brew has been removed and rewritten for some extra profits besides improved caching. Quick example:
+
+```{r}
+str(Pandoc.brew(text ='
++   Pi equals to `<%= pi %>`.
++   And here are some random data:
++   `<%= runif(10) %>`
++ '))
+```
+
+The package bundles some examples for `Pandoc.brew` to let you check its features pretty fast. 
+To brew these examples on your machine, try to run the followings commands:
+
+```
+Pandoc.brew(system.file('examples/minimal.brew', package='pander'))
+Pandoc.brew(system.file('examples/minimal.brew', package='pander'),
+            output = tempfile(), convert = 'html')
+
+Pandoc.brew(system.file('examples/short-code-long-report.brew', package='pander'))
+Pandoc.brew(system.file('examples/short-code-long-report.brew', package='pander'),
+                        output = tempfile(), convert = 'html')
+
+Pandoc.brew(system.file('examples/graphs.brew', package='pander'))
+Pandoc.brew(system.file('examples/graphs.brew', package='pander'),
+                        output = tempfile(), convert = 'html')
+```
+
+## General Options
```

Comment:
My only concern with this part is that whenever we add a new global option or edit the description of an already existing one, then we have to do the edits at the `options.R`, `README.brew` and now in this file as well. I'd rather just put a reference to the homepage -- so that maintenance would be easier for us in the future.